spec/classes/profile/hathitrust/apache_spec.rb

```diff
@@ -0,0 +1,124 @@
+
+      snippets = [
+        <<~EOT,
+          <Directory "/htapps/babel/imgsrv/cgi">
+            AllowOverride None
+
+            <Files "imgsrv">
+              SetHandler proxy:fcgi://127.0.0.1:31028
+            </Files>
+          </Directory>
+        EOT
+        <<~EOT
+          <DirectoryMatch "^/htapps/babel/([^/]+)/cgi">
+            Options +ExecCGI
+            SetHandler cgi-script
+          </DirectoryMatch>
+        EOT
+      ]
+
+      snippets.each do |snippet|
+        it { is_expected.to contain_concat_fragment(vhost_config).with_content(multiline2re(snippet)) }
+      end
+
+      it do
+        is_expected.to contain_file('access_compat.load')
+          .with(path: '/etc/apache2/mods-available/access_compat.load',
+                content: %r{LoadModule access_compat_module /usr/lib/apache2/modules/mod_access_compat.so})
+      end
+
+      it do
+        is_expected.to contain_file('access_compat.load symlink')
+          .with(ensure: 'link',
+                path: '/etc/apache2/mods-enabled/access_compat.load',
+                target: '/etc/apache2/mods-available/access_compat.load')
+      end
+
+      describe 'Production HT hostnames' do
+        %w[babel catalog www].each do |vhost|
+          it {
+            is_expected.to contain_apache__vhost("#{vhost}.hathitrust.org ssl").with(
+              servername: "#{vhost}.hathitrust.org",
+              ssl: true,
+              ssl_cert: '/etc/ssl/certs/www.hathitrust.org.crt',
+              ssl_key: '/etc/ssl/private/www.hathitrust.org.key',
+              ssl_chain: '/etc/ssl/certs/incommon_sha2.crt',
+            )
+          }
+        end
+      end
+
+      context 'with a domain and prefix specified' do
+        let(:params) do
+          {
+            domain: 'example.org',
+            prefix: 'foo.',
+          }
+        end
+
+        it { is_expected.to contain_apache__vhost('foo.babel.example.org ssl').with_servername('foo.babel.example.org') }
+        it { is_expected.to contain_apache__vhost('foo.catalog.example.org ssl').with_servername('foo.catalog.example.org') }
+        it { is_expected.to contain_apache__vhost('foo.www.example.org ssl').with_servername('foo.www.example.org') }
+
+        it {
+          is_expected.to contain_apache__vhost('foo.babel.example.org non-ssl').with(
+            redirect_dest: 'https://foo.babel.example.org',
+            servername: 'foo.babel.example.org',
+          )
+        }
+      end
+
+      context 'with a domain and no prefix specified' do
+        let(:params) do
+          {
+            domain: 'example.org',
+          }
+        end
+
+        it { is_expected.to contain_apache__vhost('babel.example.org ssl').with_servername('babel.example.org') }
+
+        context 'and domain aliases in hiera data' do
+          let(:hiera_config) { 'spec/fixtures/hiera/hathitrust_config.yaml' }
```

Comment:
I guess the reason to have the default value in data/common.yaml seems primarily for tests where we want to have some default that will at least make the tests pass. Might it make more sense to configure the tests to use this hiera fixture by default rather than putting the nonsense values in data/common.yaml?

Reply:
Oh my god, I **love** this. I didn't know it was an option, and I've really disliked the hidden features embedded in the module-level hieradata that could easily mess us up. I thought the only way around this was to add fake facts and then extra data files (like the stuff under `datacenter` and `node` in lensoftruth), which adds its own level of unwanted complexity.

Now that I know this is possible, I'd love to move to a model where the only things in common.yaml are parameters where we actually want defined behavior when undefined in our control repo. Excellent.

Reply:
I've moved all the parameters for hathitrust stuff to `hathitrust.yaml`. As we touch other things we can move them out of `data/common.yaml` as well.